test/IRGen/constant_struct_with_padding.sil

```diff
@@ -0,0 +1,26 @@
+// RUN: %swift -module-name main %s -emit-ir -o - | %FileCheck %s
```

Comment:
I guess in theory there might be a platform where i32 has some other alignment. (Heck, [ppc32 used 4-byte alignment for i1!](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/LowLevelABI/100-32-bit_PowerPC_Function_Calling_Conventions/32bitPowerPC.html#//apple_ref/doc/uid/TP40002438-SW20))